### PR TITLE
Add bump-beats pipeline

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes
Mostly copied from apm-server.

I have no access to vault secrets so I don't know how to test this for now.

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Review by oren-zohar — CHANGES_REQUESTED** [review]
@orestisfl ~~did you test it?~~ oh nvm I saw your comment about the vault secrets right now, let's sync offline

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ repos:
       - id: shfmt
         args: ["-i", "4"]
       - id: shellcheck
-        files: (deploy/azure/.*sh|scripts/publish_cft.sh)
+        files: (deploy/azure/.*sh|scripts/publish_cft.sh|.ci/scripts/update-beats.sh)
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply .pre-commit-config.yaml:78]: These are the files that the pipeline actually runs on, not the ones that are excluded.

File: .github/workflows/update-beats.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      fail-fast: false
+      matrix: ${{ fromJson(needs.filter.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+      - uses: elastic/apm-pipeline-library/.github/actions/updatecli@current
+        env:
+          BRANCH_NAME: ${{ matrix.branch }}
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply .github/workflows/update-beats.yml:36]: Probably to update dependencies for older versions as well, but perhaps it's less important for us and we can skip the extra complexity and only run it on main?

File: .github/workflows/update-beats.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+jobs:
+  filter:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    outputs:
+      matrix: ${{ steps.generator.outputs.matrix }}
+    steps:
+      - id: generator
+        uses: elastic/apm-pipeline-library/.github/actions/elastic-stack-snapshot-branches@current
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply .github/workflows/update-beats.yml]: @gurevichdmitry The bump-golang job does use vault secrets: https://github.com/elastic/cloudbeat/actions/runs/6693205116/job/18183872641

**Review by oren-zohar — APPROVED** [review]

**Comment by v1v** [comment]
> I have no access to vault secrets so I don't know how to test this for now.


You can test it locally by overriding the variables:

```
JOB_URL=noop \
    GIT_USER=noop \
    GIT_EMAIL=noop \
    GITHUB_TOKEN=your-token \
     updatecli apply --config .ci/bump-beats.yml --push=false --debug
```

**Comment by orestisfl** [comment]
> You can test it locally by overriding the variables:

I did, but I don't think it's possible to test the GH (not updatecli) action without access.

**Pull request merged**